pkg/providers/instancetype/suite_test.go

```diff
@@ -489,6 +491,34 @@ var _ = Describe("InstanceType Provider", func() {
 			node := ExpectScheduled(ctx, env.Client, pod)
 			Expect(node.Labels).To(HaveKeyWithValue(v1.LabelInstanceTypeStable, "Standard_D2_v2"))
 		})
+
+		Context("on SkuNotUnavailable, should cache SKU as unavailable in all zones", func() {
+			AssertUnavailable := func(sku string, capacityType string) {
+				// fake a SKU not available error
+				azureEnv.VirtualMachinesAPI.VirtualMachinesBehavior.VirtualMachineCreateOrUpdateBehavior.Error.Set(
+					&azcore.ResponseError{ErrorCode: instance.SKUNotAvailableErrorCode},
+				)
```

Comment:
In the future this should be replaced with more capable helpers requesting specific errors for given SKUs. Should also likely include a valid http.Response ...